### ISSUE TITLE
xcode 7.0beta could not connect to development server.

### ISSUE THREAD

**Issue by samfriend** [facebook/react-native#1563]
Today I downloaded IOS9 and xcode7.0beta which is needed for ios9

When I try to build my existing project to both simulator and iphone. It refused to load the index.ios.bundle with msg...

'Application Transport Security has blocked a cleartext HTTP (http://) resource load since it is insecure. Temporary exceptions can be configured via your app's Info.plist file.'

How to make this 'Temporary exceptions' in info.plist file? 

thank you very much

**Comment by samfriend** [comment]
Adding the following to your Info.plist will disable ATS
 
<key>NSAppTransportSecurity</key>  
     <dict>  
          <key>NSAllowsArbitraryLoads</key><true/>  
     </dict>

**Comment by JohnyDays** [comment]
I think this issue shouldn't be closed, we should probably try to make this work out of the box. Or perhaps apple should allow localhost or local ips to be loaded regardless?

**Comment by ide** [comment]
File a bug with Apple if you have a case for local IPs. They might already have a way to whitelist certain IPs in which case we could add localhost. I don't think weakening security for the entire app is a good default if we can avoid it.

**Comment by JohnyDays** [comment]
I'd imagine they implemented this to prevent naive implementations of OTA updates. But I do think it doesn't make sense for local IPs. I'm not an expert on IP addresses and other network-y things, but there's a few select ip ranges that are reserved for local networks right? Perhaps that could be automatically added to the whitelist(or atleast referenced somewhere)

**Comment by ide** [comment]
It has nothing to do with OTA updates. This is blocking HTTP and requiring HTTPS. An alternative solution would be for the packager to serve a self-signed certificate though I'm not sure what it would take to get the client to accept it.

**Comment by dvcrn** [comment]
The info.plist entry above solves the problem inside the simulator for me, however on the actual device I still get 'could not connect to development server'. Does anyone have a similar problem?

**Comment by danielkramer** [comment]
I've added the plist entry but it's not clearing up the error.  I'm using the latest Xcode Beta released this week  - Version 7.0 beta 3 (7A152u)

**Comment by sergeyu** [comment]
Works for me  for Version 7.0 beta 3 (7A152u)

**Comment by ide** [comment]
This is fixed in master (create a new project to get the changes).

**Comment by gunn** [comment]
So is there any way to update an existing project? (other than create a new
one and copy files)

**Comment by ide** [comment]
You could look at the SampleApp's xcodeproj and manually port over the plist entry that's mentioned earlier in this thread.

**Comment by kbpontius** [comment]
The above .plist entry worked immediately for me (on an old project ported to Swift 2).

**Comment by DIJOiGloo** [comment]
I've added the plist entry but it's not clearing up the error. I'm using the latest Xcode Beta released this week - Version 7.1 beta (7B75)

**Comment by edo1493** [comment]
I am in the same situation, which is quite annoying since I have been working on this project for the last three months. However I am not sure if the simulator doesn't pick up the new updated info.plst or that is not working.

**Comment by edo1493** [comment]
I am sorted, I was trying to change the info from xcode, but opening the file clearly showed my xml was wrong.

**Comment by DIJOiGloo** [comment]
In my case, i can fixed it.
![screen shot 2015-10-01 at 9 16 24 am](https://cloud.githubusercontent.com/assets/14836210/10209288/2ba74a94-681d-11e5-8a07-76d5fc1c58bf.png)

**Comment by fatinWasta** [comment]
using xcode 7.1 beta 3. None of the solutions are working for me. Did any one solved for xcode 7.1 beta 3?

**Comment by yychun1217** [comment]
tried, the above solutions not working for me either. Any suggestion?

**Comment by DIJOiGloo** [comment]
Yes I can

**Comment by DIJOiGloo** [comment]
Am using Xcode Version 7.1 beta (7B75)

**Comment by niftylettuce** [comment]
What's the solution here?  I read the app faces rejection possibility if this is in your plist:

**Comment by niftylettuce** [comment]
https://github.com/mjohnston/react-native-webpack-server/pull/61

**Comment by niftylettuce** [comment]
Looks like the proper fix is:

```plist
    <key>NSAppTransportSecurity</key>
    <dict>
        <key>NSExceptionDomains</key>
        <dict>
            <key>localhost</key>
            <dict>
                <key>NSTemporaryExceptionAllowsInsecureHTTPSLoads</key>
                <false/>            
                <key>NSIncludesSubdomains</key>
                <true/>
                <key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
                <true/>
                <key>NSTemporaryExceptionMinimumTLSVersion</key>
                <string>1.0</string>
                <key>NSTemporaryExceptionRequiresForwardSecrecy</key>
                <false/>
            </dict>
        </dict>
    </dict> 
```

**Comment by DIJOiGloo** [comment]
Hi,

Definitely  apple reject this this is because. http request does not accept apple we need to change https request,

in My case also rejected :)


DIJO JOSEPH
dijo@igloosoftware.com <mailto:dijo@igloosoftware.com>.au
0452505282
"If you want to shine like a sun. First burn like a sun.” - A.P.J Abdul Kalam

**Comment by niftylettuce** [comment]
According to https://forums.developer.apple.com/thread/22710 it ignores port number too, so that won't work either (affixing `:12345`) to your plist definition.

**Comment by niftylettuce** [comment]
Aha, I fixed it, for anyone else having this issue, make sure you don't have more than one key for `NSAppTransportSecurity`.

**Comment by Nandulucky** [comment]
I fixed it in Xcode 7.2, App Transport Security Settings as Dictionary and Allow Arbitrary Loads as YES.
it's working absolutely.

**Comment by bradleywhite2014** [comment]
If you're using React Native, niftylettuce has the correct fix.
